src/librustc/session/config.rs

```diff
@@ -695,6 +722,17 @@ pub fn default_configuration(sess: &Session) -> ast::CrateConfig {
     if sess.opts.debug_assertions {
         ret.push(attr::mk_word_item(InternedString::new("debug_assertions")));
     }
+
+    sess.opts.cg.sanitize.map(|s| {
+        let name = match s {
+            Sanitize::Address => "sanitize_address",
+            Sanitize::Leak    => "sanitize_leak",
+            Sanitize::Memory  => "sanitize_memory",
+            Sanitize::Thread  => "sanitize_thread"
+        };
+        ret.push(attr::mk_word_item(InternedString::new(name)));
+    });
```

Comment:
Ok, that seems reasonable yeah. And yes you can check or define both of `sanitize = "foo"` and `sanitize = "bar"` in the same program (this is how Cargo features work)